src/targets/services/deleteOldShares.js

```diff
@@ -0,0 +1,85 @@
+global.fetch = require('node-fetch').default
+
+import CozyClient, { Q } from 'cozy-client'
+import { createLogger, getAppDirectory } from './helpers'
+
+import config from '../../../dissec.config.json'
+
+/**
+ * This service can be launched in two ways:
+ * - Periodically to erase old shares
+ * - Occasionally to remove specific shares
+ *
+ * When triggered, it requires a list of file ids
+ */
+export const deleteOldShares = async () => {
+  const client = CozyClient.fromEnv(process.env, {})
+  const log = createLogger(client.stackClient.uri)
+  const appDirectory = await getAppDirectory(client)
+
+  // Retrieve arguments stored in the jobs
+  const jobId = process.env['COZY_JOB_ID'].split('/')[2]
+  const job = await client.query(Q('io.cozy.jobs').getById(jobId))
+  log(`Deleting old shares. Job ID: ${jobId}`)
+
+  if (job.data.attributes.message.metadata) {
```

Comment:
I think it would be worth explaining in which situation you have such metadata, or not